numba/cpython/hashing.py

```diff
@@ -76,7 +82,15 @@ def _Py_HashDouble(v):
             else:
                 return -_PyHASH_INF
         else:
-            return _PyHASH_NAN
+            # Python 3.10 does not use `_PyHASH_NAN`.
+            # https://github.com/python/cpython/blob/2c4792264f9218692a1bd87398a60591f756b171/Python/pyhash.c#L102   # noqa: E501
+            # Numba returns a pseudo-random number to reflect the spirit of the
+            # change.
+            if _py310_or_later:
+                x = _prng_random_hash()
```

Comment:
I checked the performance difference here. With the py310 change, it takes about ~9ns. Before, it takes about ~3ns. I was worried that we might be looking a >10x difference. The 2-3x difference for a rare case is acceptable IMO.

Reply:
Thanks for checking @sklam